### ISSUE TITLE
Make all calls through `execute`

### ISSUE THREAD

**Issue by martriay** [argentlabs/argent-contracts-starknet#4]
What do you think of making all user calls go through `execute`, even if they're account operations. [Like this](https://github.com/OpenZeppelin/cairo-contracts/commit/113d24778b6f372c8989336e1ec7b3d4ec21e3f0#diff-d223c9adb70603fcfe336e89fb620a6971d5ddf101a181e9c06ab077d2821e7aR59-R126). That way, you don't have to reimplement signature validation, nonce management, etc. It's leaner and also suggest a single and simple entrypoint for all user interactions.

Then you can do something like [this](https://github.com/OpenZeppelin/cairo-contracts/blob/main/test/Account.py#L76):
```python
await signer.send_transaction(account, account.contract_address, 'set_L1_address', [ANOTHER_ADDRESS])
```

**Comment by juniset** [comment]
I've considered it but I don't think it is a good fit because we have different signature requirements depending on the method being called. So you would need extra logic in `execute` to first get the requirement for the target method (e.g. you need 2 signatures) before you can verify and execute. This is exactly what we do in the Argent Vault on L1 but I think it would add an unnecessary overhead here given the minimal specification.

**Comment by juniset** [comment]
Fixed in [#17].
